### DISCUSSION TITLE
Image type resizing in a repeatable field ?

### DISCUSSION THREAD

**Discussion by t-prod** [Laravel-Backpack/community-forum#257] [Q&A (Help)]
Hi,

In a repeatable field and specifically in a subfield type image, how can we handle the setImageAttribute ?
I would like to resize image

Regards

**Comment by jcastroa87** [comment]
To handle the setImageAttribute for an image subfield within a repeatable field in Backpack, and to resize the image, you can follow these steps:

First, ensure you have set up the repeatable field with an image subfield correctly in your CRUD controller:
```
CRUD::field([
    'name'      => 'images',
    'type'      => 'repeatable',
    'subfields' => [
        [
            'name'    => 'image',
            'type'    => 'image',
            'label'   => 'Image',
            'withFiles' => [
                'disk' => 'public',
                'path' => 'your-path',
            ],
        ],
        // Other subfields...
    ],
]);
```

In your model, you'll need to create a mutator for the images attribute. Since the repeatable field stores data as JSON, you'll need to decode it, process each image, and then encode it back to JSON.
Here's an example of how you can implement the mutator to resize images:

```
use Illuminate\Support\Facades\Storage;
use Intervention\Image\Facades\Image;

public function setImagesAttribute($value)
{
    $attribute_name = "images";
    $disk = "public";
    $destination_path = "your-path";

    $images = json_decode($value);

    foreach ($images as &$item) {
        if (isset($item->image) && starts_with($item->image, 'data:image')) {
            // Make the image
            $image = Image::make($item->image);
            
            // Resize the image (adjust dimensions as needed)
            $image->resize(300, 300, function ($constraint) {
                $constraint->aspectRatio();
                $constraint->upsize();
            });

            // Generate a filename
            $filename = md5($item->image . time()) . '.jpg';

            // Store the image on disk
            Storage::disk($disk)->put($destination_path . '/' . $filename, $image->stream());

            // Save the path to the item
            $item->image = $destination_path . '/' . $filename;
        }
    }

    $this->attributes[$attribute_name] = json_encode($images);
}
```

This implementation will resize each image in the repeatable field to 300x300 pixels while maintaining the aspect ratio. You can adjust the dimensions as needed.
Remember to install and set up the Intervention Image library if you haven't already, as it's used for image manipulation in this example.
[Laravel-Backpack/community-forum #848](https://github.com/Laravel-Backpack/community-forum/discussions/848) provides additional context on handling file uploads in repeatable fields.
Note that this approach assumes you're using base64 encoded images. If you're using file uploads, you might need to adjust the logic slightly to handle file objects instead of base64 strings.
